examples/cp4ba/main.tf

```diff
@@ -18,36 +18,32 @@ resource "null_resource" "mkdir_kubeconfig_dir" {
 
 data "ibm_container_cluster_config" "cluster_config" {
   depends_on        = [null_resource.mkdir_kubeconfig_dir]
-  cluster_name_id   = var.cluster_name_or_id
+  cluster_name_id   = var.cluster_id
   resource_group_id = data.ibm_resource_group.resource_group.id
   config_dir        = local.cluster_config_path
 }
 
-module "cp4ba" {
+module "install_cp4ba" {
   source = "../../modules/cp4ba"
 //  source = "git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak/tree/main/modules/cp4ba"
-  enable = true
-
-  # ---- Cluster settings ----
-  cluster_config_path = data.ibm_container_cluster_config.cluster_config.config_file_path
-  ingress_subdomain = var.ingress_subdomain
-
-  # ---- Cloud Pak settings ----
-  cp4ba_project_name      = "cp4ba"
-  entitled_registry_user  = var.entitled_registry_user
-  entitlement_key         = var.entitlement_key
-
+  enable_cp4ba           = var.enable_cp4ba
+  ibmcloud_api_key       = var.ibmcloud_api_key
+  region                 = var.region
+  cluster_config_path    = data.ibm_container_cluster_config.cluster_config.config_file_path
+  ingress_subdomain      = var.ingress_subdomain
+  # ---- Platform ----
+  cp4ba_project_name     = var.cp4ba_project_name
+  entitled_registry_user_email = var.entitled_registry_user_email
+  entitled_registry_key        = var.entitled_registry_key
+  # ----- LDAP Settings -----
+  ldap_admin_name         = var.ldap_admin_name
+  ldap_admin_password     = var.ldap_admin_password
   # ----- DB2 Settings -----
-  db2_host_name = var.db2_host_name
-  db2_host_port = var.db2_host_port
-  db2_admin     = var.db2_admin
-  db2_user      = var.db2_user
-  db2_password  = var.db2_password
+  db2_host_port           = var.db2_host_port # != null ? var.db2_ports : module.install_db2.db2_ports # var.db2_port_number
+  db2_host_address        = var.db2_host_address
+  db2_admin_username      = var.db2_admin_username
+  db2_admin_user_password = var.db2_admin_user_password
 
-  # ----- LDAP Settings -----
-  ldap_admin    = var.ldap_admin
-  ldap_password = var.ldap_password
-  ldap_host_ip  = var.ldap_host_ip
 }
```

Comment:
`ldap_host_ip` is still required by the cp4ba module. Did you test these changes?  This should have given an error.

Reply:
Yes, I restructured the scripts. `ldap_host_ip` is still there. 
```
  cp4ba_deployment_file_content             = templatefile("${path.module}/templates/cp4ba_deployment.yaml.tmpl", {
    ldap_host_ip     = var.ldap_host_ip,
    db2_host_address    = var.db2_host_address,
    db2_host_port    = var.db2_host_port,
    ingress_subdomain = var.ingress_subdomain
  })
```